pkg/labels/matcher.go

```diff
@@ -199,3 +204,13 @@ func (ms Matchers) String() string {
 
 	return buf.String()
 }
+
+// This is copied from matchers/parse/lexer.go. It will be removed when
+// the transition window from classic matchers to UTF-8 matchers is complete,
+// as then we can use double quotes when printing the label name for all
+// matchers. Until then, the classic parser does not understand double quotes
+// around the label name, so we use this function as a heuristic to tell if
+// the matcher was parsed with the UTF-8 parser or the classic parser.
+func isReserved(r rune) bool {
```

Comment:
I'm assuming making the function public and re-using it here was a no-go for some reason?

I'd love if we could please keep track of these tidbits that we need to remove once we're done.

Reply:
I added a task to https://github.com/prometheus/alertmanager/issues/3486

Reply:
It's a bit of an odd one I think. Remember how we made the lexer and all it's related functions package-private? It felt weird to me that we would expose this **one** function that checks for reserved characters when the rest of the lexer and it's related functions are inaccessible.

Besides, I'm thinking we'll delete this once we removed the old parser so I felt it was cleaner to duplicate it and add a comment.